src/DonationForms/resources/app/hooks/useDonationSummary.ts

```diff
@@ -1,22 +1,118 @@
 import {
-    DonationSummaryLineItem,
+    DonationSummaryLineItem, DonationTotals,
     useDonationSummaryContext,
-    useDonationSummaryDispatch,
+    useDonationSummaryDispatch
 } from '@givewp/forms/app/store/donation-summary';
 import {
     addAmountToTotal,
     addItem,
     removeAmountFromTotal,
-    removeItem,
+    removeItem
 } from '@givewp/forms/app/store/donation-summary/reducer';
 import {useCallback} from '@wordpress/element';
+import type {
+    subscriptionPeriod
+} from '@givewp/forms/registrars/templates/groups/DonationAmount/subscriptionPeriod';
 
 /**
+ * Zero decimal currencies are currencies that do not have a minor unit.
+ * For example, the Japanese Yen (JPY) does not have a minor unit.
+ * @unreleased
+ *
+ * @see https://stripe.com/docs/currencies#zero-decimal
+ */
+const zeroDecimalCurrencies = [
+    'BIF',
+    'CLP',
+    'DJF',
+    'GNF',
+    'JPY',
+    'KMF',
+    'KRW',
+    'MGA',
+    'PYG',
+    'RWF',
+    'UGX',
+    'VND',
+    'VUV',
+    'XAF',
+    'XOF',
+    'XPF',
+];
+
+/**
+ * Takes in an amount value in dollar units and returns the calculated cents (minor) amount
+ *
+ * @unreleased
+ */
+const dollarsToCents = (amount: string, currency: string) => {
```

Comment:
Nit: I suggest `amountToMinorUnit` — as "dollars" and "cents" are distinct terms for certain currencies. 🤓